dora/services/views.py

```diff
@@ -667,30 +674,32 @@ def search(request):
             "subcategories",
         )
     )
-    if category:
-        services = services.filter(categories__value=category)
+    if categories:
+        services = services.filter(categories__value__in=categories.split(","))
 
     if kinds:
         services = services.filter(kinds__value__in=kinds.split(","))
 
-    if subcategory:
-        cat, subcat = subcategory.split("--")
-        if subcat == "other":
-            # Quand on cherche une sous-catégorie de type 'Autre', on veut
-            # aussi remonter les services sans sous-catégorie
-            all_sister_subcats = ServiceSubCategory.objects.filter(
-                value__startswith=f"{cat}--"
-            )
-            services = services.filter(
-                Q(subcategories__value=subcategory)
-                | (Q(categories__value=cat) & ~Q(subcategories__in=all_sister_subcats))
-            )
-        else:
-            services = services.filter(subcategories__value=subcategory)
+    if subcategories:
+        subcategories_filter = Q()
+        for subcategory in subcategories.split(","):
+            cat, subcat = subcategory.split("--")
+            if subcat == "other":
```

Comment:
attention: reprendre la correction sur main: 
```suggestion
            if subcat == "autre":
```